From a6d3b7f3c83cdbb9c6cb150528d6b48dce80cda0 Mon Sep 17 00:00:00 2001
From: Mike Panetta <panett...@gmail.com>
Date: Thu, 27 Jun 2013 07:06:03 -0500
Subject: [PATCH] imx6dl-wandboard: Updated device tree patch for 3.9.0-rc6+,
 diff is against the for-next branch.

I seem to have managed to get most of the IO working on my wandboard with
the 3.9.0-rc6+ kernel.  I am not 100% sure how since I have not actually
set up the IO pins that reset and power the wireless chip, but even
wireless is working (well I get an address via DHCP anyway).  Here is the
patch, cut and pasted from a terminal (if it does not work I can attach a
file later).

NOTE:  This patch does not have anything in it to enable the HDMI port or
any of the audio yet.  It isn't immediately important to me so I did not do
it yet.
---
 arch/arm/boot/dts/imx6dl-wandboard.dts |   64 +++++++++++++++++++++++
 arch/arm/boot/dts/imx6dl.dtsi          |   87 ++++++++++++++++++++++++++++++--
 2 files changed, 148 insertions(+), 3 deletions(-)

diff --git a/arch/arm/boot/dts/imx6dl-wandboard.dts b/arch/arm/boot/dts/imx6dl-wandboard.dts
index bfc59c3..3846f5b 100644
--- a/arch/arm/boot/dts/imx6dl-wandboard.dts
+++ b/arch/arm/boot/dts/imx6dl-wandboard.dts
@@ -18,6 +18,16 @@
 	memory {
 		reg = <0x10000000 0x40000000>;
 	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		debug-led {
+			label = "Heartbeat";
+			gpios = <&gpio3 27 0>;
+			linux,default-trigger = "heartbeat";
+		};
+	};
 };
 
 &fec {
@@ -27,16 +37,70 @@
 	status = "okay";
 };
 
+&iomuxc {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_hog>;
+
+	hog {
+		pinctrl_hog: hoggrp {
+			fsl,pins = <
+				MX6DL_PAD_EIM_DA13__GPIO3_IO13  0x80000000 /* BT_ON */
+				MX6DL_PAD_EIM_A25__GPIO5_IO02   0x80000000 /* WL_ON */
+				MX6DL_PAD_EIM_DA15__GPIO3_IO15  0x80000000 /* BT_HOST_WAKE */
+				MX6DL_PAD_ENET_TXD1__GPIO1_IO29 0x80000000 /* WL_HOST_WAKE */
+				MX6DL_PAD_EIM_DA11__GPIO3_IO11  0x80000000 /* EIM_DA11/GPIO3_12 */
+				MX6DL_PAD_EIM_D27__GPIO3_IO27   0x80000000 /* GPIO3_27 */
+			>;
+		};
+	};
+};
+
+&i2c1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_i2c1_1>;
+	status = "okay";
+};
+
+&i2c2 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_i2c2_2>;
+	status = "okay";
+};
+
+&i2c3 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_i2c3_2>;
+	status = "okay";
+};
+
 &uart1 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_uart1_1>;
 	status = "okay";
 };
 
+&uart3 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart3_2>;
+	status = "okay";
+};
+
 &usbh1 {
 	status = "okay";
 };
 
+&usdhc1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_usdhc1_1>;
+	status = "okay";
+};
+
+&usdhc2 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_usdhc2_2>;
+	status = "okay";
+};
+
 &usdhc3 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_usdhc3_2>;
diff --git a/arch/arm/boot/dts/imx6dl.dtsi b/arch/arm/boot/dts/imx6dl.dtsi
index 5bcdf3a..1c3306d 100644
--- a/arch/arm/boot/dts/imx6dl.dtsi
+++ b/arch/arm/boot/dts/imx6dl.dtsi
@@ -78,6 +78,47 @@
 					};
 				};
 
+				i2c1 {
+					pinctrl_i2c1_1: i2c1grp-1 {
+						fsl,pins = <
+							MX6DL_PAD_EIM_D21__I2C1_SCL 0x4001b8b1
+							MX6DL_PAD_EIM_D28__I2C1_SDA 0x4001b8b1
+						>;
+					};
+				};
+
+				i2c2 {
+					pinctrl_i2c2_1: i2c2grp-1 {
+						fsl,pins = <
+							MX6DL_PAD_EIM_EB2__I2C2_SCL  0x4001b8b1
+							MX6DL_PAD_EIM_D16__I2C2_SDA  0x4001b8b1
+						>;
+					};
+
+					pinctrl_i2c2_2: i2c2grp-2 {
+						fsl,pins = <
+							MX6DL_PAD_KEY_COL3__I2C2_SCL 0x4001b8b1
+							MX6DL_PAD_KEY_ROW3__I2C2_SDA 0x4001b8b1
+						>;
+					};
+				};
+
+				i2c3 {
+					pinctrl_i2c3_1: i2c3grp-1 {
+						fsl,pins = <
+							MX6DL_PAD_EIM_D17__I2C3_SCL 0x4001b8b1
+							MX6DL_PAD_EIM_D18__I2C3_SDA 0x4001b8b1
+					>;
+				};
+
+					pinctrl_i2c3_2: i2c3grp-2 {
+						fsl,pins = <
+							MX6DL_PAD_GPIO_5__I2C3_SCL  0x4001b8b1
+							MX6DL_PAD_GPIO_16__I2C3_SDA 0x4001b8b1
+						>;
+					};
+				};
+
 				uart1 {
 					pinctrl_uart1_1: uart1grp-1 {
 						fsl,pins = <
@@ -87,6 +128,24 @@
 					};
 				};
 
+				uart3 {
+					pinctrl_uart3_1: uart3grp-1 {
+						fsl,pins = <
+							MX6DL_PAD_EIM_D24__UART3_TX_DATA    0x1b0b1
+							MX6DL_PAD_EIM_D25__UART3_RX_DATA    0x1b0b1
+						>;
+					};
+
+					pinctrl_uart3_2: uart3grp-2 {
+						fsl,pins = <
+							MX6DL_PAD_EIM_D24__UART3_TX_DATA    0x1b0b1
+							MX6DL_PAD_EIM_D25__UART3_RX_DATA    0x1b0b1
+							MX6DL_PAD_EIM_D23__UART3_CTS_B      0x1b0b1
+							MX6DL_PAD_EIM_EB3__UART3_RTS_B      0x1b0b1
+						>;
+					};
+				};
+
 				uart4 {
 					pinctrl_uart4_1: uart4grp-1 {
 						fsl,pins = <
@@ -104,6 +163,19 @@
 					};
 				};
 
+				usdhc1 {
+					pinctrl_usdhc1_1: usdhc1grp-1 {
+						fsl,pins = <
+							MX6DL_PAD_SD1_CMD__SD1_CMD    0x17059
+							MX6DL_PAD_SD1_CLK__SD1_CLK    0x10059
+							MX6DL_PAD_SD1_DAT0__SD1_DATA0 0x17059
+							MX6DL_PAD_SD1_DAT1__SD1_DATA1 0x17059
+							MX6DL_PAD_SD1_DAT2__SD1_DATA2 0x17059
+							MX6DL_PAD_SD1_DAT3__SD1_DATA3 0x17059
+						>;
+					};
+				};
+
 				usdhc2 {
 					pinctrl_usdhc2_1: usdhc2grp-1 {
 						fsl,pins = <
@@ -119,6 +191,17 @@
 							MX6DL_PAD_NANDF_D7__SD2_DATA7 0x17059
 						>;
 					};
+
+					pinctrl_usdhc2_2: usdhc2grp-2 {
+						fsl,pins = <
+							MX6DL_PAD_SD2_CMD__SD2_CMD    0x17059
+							MX6DL_PAD_SD2_CLK__SD2_CLK    0x10059
+							MX6DL_PAD_SD2_DAT0__SD2_DATA0 0x17059
+							MX6DL_PAD_SD2_DAT1__SD2_DATA1 0x17059
+							MX6DL_PAD_SD2_DAT2__SD2_DATA2 0x17059
+							MX6DL_PAD_SD2_DAT3__SD2_DATA3 0x17059
+						>;
+					};
 				};
 
 				usdhc3 {
@@ -137,7 +220,7 @@
 						>;
 					};
 
-					pinctrl_usdhc3_2: usdhc3grp_2 {
+					pinctrl_usdhc3_2: usdhc3grp-2 {
 						fsl,pins = <
 							MX6DL_PAD_SD3_CMD__SD3_CMD    0x17059
 							MX6DL_PAD_SD3_CLK__SD3_CLK    0x10059
@@ -148,8 +231,6 @@
 						>;
 					};
 				};
-
-
 			};
 
 			pxp: pxp@020f0000 {
-- 
1.7.10.4

